oss_src/unity/python/sframe/cython/cy_cpp_utils.pyx

```diff
@@ -52,3 +52,28 @@ cdef string _attempt_cast_str_to_cpp(py_s) except *:
 
     # Okay, none of these worked, so error out.
     raise TypeError("Type '%s' cannot be interpreted as str." % str(type(py_s)))
+
+
+cdef bint _cpp_to_str_py3_decode_enabled = True
+
+cpdef disable_cpp_str_decode():
+    global _cpp_to_str_py3_decode_enabled
+    assert _cpp_to_str_py3_decode_enabled == True
+    _cpp_to_str_py3_decode_enabled = False
+
+cpdef enable_cpp_str_decode():
+    global _cpp_to_str_py3_decode_enabled
+    assert _cpp_to_str_py3_decode_enabled == False
+    _cpp_to_str_py3_decode_enabled = True
+
```

Comment:
Yes, the guideline is to use this strictly in a try ... finally block.  But
let's do implement it as a with mechanism.  I'll revisit that.

<http://www.dato.com>   Hoyt Koepke
Data Scientist
hoytak@dato.com   <https://twitter.com/datoinc>
<http://www.linkedin.com/company/graphlab>
<https://www.facebook.com/datoinc>
<http://www.youtube.com/channel/UC9T07WMdwB6IPf_dKXhVsCQ>


On Mon, Mar 14, 2016 at 2:44 PM, Yucheng Low <notifications@github.com>
wrote:

> In oss_src/unity/python/sframe/cython/cy_cpp_utils.pyx
> <https://github.com/dato-code/SFrame/pull/216#discussion_r56081629>:
>
> > @@ -52,3 +52,28 @@ cdef string _attempt_cast_str_to_cpp(py_s) except *:
> >
> >      # Okay, none of these worked, so error out.
> >      raise TypeError("Type '%s' cannot be interpreted as str." % str(type(py_s)))
> > +
> > +
> > +cdef bint _cpp_to_str_py3_decode_enabled = True
> > +
> > +cpdef disable_cpp_str_decode():
> > +    global _cpp_to_str_py3_decode_enabled
> > +    assert _cpp_to_str_py3_decode_enabled == True
> > +    _cpp_to_str_py3_decode_enabled = False
> > +
> > +cpdef enable_cpp_str_decode():
> > +    global _cpp_to_str_py3_decode_enabled
> > +    assert _cpp_to_str_py3_decode_enabled == False
> > +    _cpp_to_str_py3_decode_enabled = True
> > +
>
> note that this is kind of iffy. Especially since the flag could be set,
> then a ctrl-C / some other failure happens which cause the flag to stay
> set. A "with xxx" mechanism is preferred.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dato-code/SFrame/pull/216/files#r56081629>.
>